images/airflow/2.10.3/python/mwaa/celery/sqs_broker.py

```diff
@@ -1150,7 +1150,10 @@ def _size(self, queue):
         resp = c.get_queue_attributes(
             QueueUrl=url, AttributeNames=["ApproximateNumberOfMessages"]
         )
-        return int(resp["Attributes"]["ApproximateNumberOfMessages"])
+        if "Attributes" in resp:
+            return int(resp["Attributes"]["ApproximateNumberOfMessages"])
+        else:
+            logger.error("Unexpected response from SQS get_queue_attributes: %s", resp)
```

Comment:
In error cases, this will cause exception/un-intended behaviour in calling function which may cause confusion among existing customers. Instead of if else, lets catch exception, log it and rethrow